### PR TITLE
routes map

### PR DESCRIPTION
**Pull request by lachlan89**
Hey guys - please look over this routing plan in server/api/routes.md and provide any feedback. Thanks!

### CODE REVIEW

File: server/api/routes.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+**USER: (/api/user)**
+
+* register guest user (POST) - occurs as soon as any user visits site (any page). Sets cookie (sessionId)
+* create user account (POST)
+* authenticate registered user (POST) - sets cookie (some version of userId)
```
Comment (ajmorozoff) [inline server/api/routes.md]: I think authentication should use the /auth routes and not overload the /api/user route
We should follow the pattern Eliot first introduced in the auth lecture  with a GET to a /auth/whoami route

File: server/api/routes.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+* retrieve single user, or all users (GET)
+* update user (PUT)
+* delete user account (DELETE)
+
+**CART/CART-ITEM: (/api/cart)**
```
Comment (ajmorozoff) [inline server/api/routes.md]: Let's have a separate section for CART and CART-ITEM models?

I suggest we might use
GET /api/cart/:cartId to return a cart and  ALL cart items
GET /api/cartItem/:itemId  to return a single cart item (this probably won't be leveraged directly by a call from the frontend)

File: server/api/routes.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* create new cart (POST) - use userId (new cart should be created as soon as new user enters app)
+* retrieve single cart, or all carts (GET)
+* add to cart (POST) - creates new CartItem, uses productId of selected product, cartId
+* retrieve single CartItem, or all CartItem(s) (GET)
+* edit cart item (PUT)
+* delete cart item (DELETE)
+* clear cart (PUT)
+
+**PRODUCT: (/api/products)**
+* add new product (POST) - this should create new price in the process
```
Comment (ajmorozoff) [inline server/api/routes.md]: The thunk will create the price, right?

File: server/api/routes.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+* edit product (PUT) - this will be used whenever a user adds a product to cart or updates quantity, as well as admin purposes
```
Comment (ajmorozoff) [inline server/api/routes.md]: A user adding a product to a cart will be the result of a POST to /cartItem
A user updating quantity of an item in the cart will be a PUT to /cartItem

File: server/api/routes.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+* delete user account (DELETE)
+
+**CART/CART-ITEM: (/api/cart)**
+* create new cart (POST) - use userId (new cart should be created as soon as new user enters app)
+* retrieve single cart, or all carts (GET)
+* add to cart (POST) - creates new CartItem, uses productId of selected product, cartId
+* retrieve single CartItem, or all CartItem(s) (GET)
+* edit cart item (PUT)
+* delete cart item (DELETE)
+* clear cart (PUT)
```
Comment (ajmorozoff) [inline server/api/routes.md]: Clear cart  could also just be a DELETE of all cartItems for a given cartId

File: server/api/routes.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+**PRODUCT: (/api/products)**
+* add new product (POST) - this should create new price in the process
+* edit product (PUT) - this will be used whenever a user adds a product to cart or updates quantity, as well as admin purposes
+* retrieve individual product, or all products (GET) - add queries to sort and filter (e.g. product in category)
+* delete product (DELETE) - admin only!
+
+**PRICE: (/api/price)**
+* create new price (POST)
+* retrieve single price, or all prices (GET)
+* update price (PUT)
```
Comment (ajmorozoff) [inline server/api/routes.md]: I'm thinking we should change the relationship between PRODUCT and PRICE models. I think we should make sure that PRICE belongs to  PRODUCT so if  I  update the price, it only updates for one given  product.

Comment (ajmorozoff) [reply server/api/routes.md]: Looks like it's currently set up so that Price has a productId. I think it's fine this  way...?

Comment (amolsaxe3) [reply server/api/routes.md]: Do we really need Price model separately? if yes please let me know what is the thought process.

Comment (ajmorozoff) [reply server/api/routes.md]: It was recommended in some database design examples as a way to simplify updating prices. 
We can collapse it into the  PRODUCT model  to start if it's easier.

Comment (amolsaxe3) [reply server/api/routes.md]: yes.

File: server/api/routes.md (hunk 1 above)
Comment (lachlan89) [reply server/api/routes.md]: 👍

File: server/api/routes.md (hunk 2 above)
Comment (lachlan89) [reply server/api/routes.md]: sure

File: server/api/routes.md (hunk 3 above)
Comment (lachlan89) [reply server/api/routes.md]: Dismissing this since we no longer have Price model

File: server/api/routes.md (hunk 4 above)
Comment (lachlan89) [reply server/api/routes.md]: yes yes absolutely

File: server/api/routes.md (hunk 5 above)
Comment (lachlan89) [reply server/api/routes.md]: true dat

**Comment by lachlan89** [comment]
Made changes!

**Pull request merged**